src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -347,11 +391,38 @@ static RuntimeModule GetModuleWorker(RuntimeType type)
 
         public ModuleHandle GetModuleHandle()
         {
+            if (m_type is null)
+            {
+                throw new ArgumentException(SR.Arg_InvalidHandle);
+            }
+
             return new ModuleHandle(GetModule(m_type));
         }
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern RuntimeType GetBaseType(RuntimeType type);
+        internal static RuntimeType? GetBaseType(RuntimeType type)
+        {
+            if (IsGenericVariable(type))
+            {
+                throw new ArgumentException(SR.Arg_InvalidHandle);
+            }
+
+            TypeHandle typeHandle = type.GetNativeTypeHandle();
+            if (typeHandle.IsTypeDesc)
+            {
+                return null;
+            }
+
+            MethodTable* pParentMT = typeHandle.AsMethodTable()->ParentMethodTable;
+            if (pParentMT == null)
+            {
+                return null;
+            }
+
+            RuntimeType? result = null;
+            GetTypeObject(pParentMT, ObjectHandleOnStack.Create(ref result));
+            GC.KeepAlive(type);
+            return result!;
```

Comment:
```suggestion
            RuntimeType result = GetTypeObject(pParentMT);
            GC.KeepAlive(type);
            return result;
```
We should use the helper with fast path.